src/onramp.rs

```diff
@@ -275,13 +289,14 @@ mod test {
 
             let module_path = &tremor_script::path::ModulePath { mounts: Vec::new() };
             let aggr_reg = tremor_script::aggr_registry();
-            let artefact = tremor_pipeline::query::Query::parse(
+            let artefact = tremor_pipeline::query::Query::parse_with_args(
                 &module_path,
                 "select event from in into out;",
                 "<test>",
                 Vec::new(),
                 &*tremor_pipeline::FN_REGISTRY.lock()?,
                 &aggr_reg,
+                &literal!({}),
```

Comment:
`parse_with_args` and args being `&literal!({})` seems to be common, would it make sense to call parse w/o args and then default args to a literal `{}`?

Reply:
Yep - we can remove the `parse` and rename `parse_with_args` to `parse` and pass through the args here I think

Reply:
I went half way and changed the above to `parse` - assuming we remove support for legacy onramps ( depending on the connectors merge + needs ) - we can cleanup. This leaves deprecate vs hard removal open for discussion closer to the release. Which is the same as defaulting args by effect.